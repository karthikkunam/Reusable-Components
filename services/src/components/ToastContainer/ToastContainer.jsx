import React, { useState } from 'react';
import PropTypes from 'prop-types';

const ToastContainer = props => {
  const [showToast, setShowToast] = useState(true);
  const { toastHeader, hideClose, children, autoClose, toastPosition } = props;

  const onClose = () => {
    const { onToastClose } = props;
    setShowToast(false);
    onToastClose && onToastClose(false);
  };
  setTimeout(() => {
    if (autoClose) {
      setShowToast(false);
    }
  }, autoClose);

  return (
    <div
      className={`toast ${toastPosition} ${showToast ? 'show' : 'hide'}`}
      role="alert"
      aria-live="assertive"
      aria-atomic="true"
      data-autohide="true"
      data-delay="3000">
      <div className="toast-header">
        <strong className="mr-auto">{toastHeader}</strong>
        {!hideClose && (
          <button type="button" className="close" data-dismiss="toast" onClick={onClose}>
            <span className="icon icon-close"></span>
          </button>
        )}
      </div>
      <div className="toast-body">{children}</div>
    </div>
  );
};

ToastContainer.defaultProps = {
  hideClose: false,
};

ToastContainer.propTypes = {
  toastHeader: PropTypes.node,
  children: PropTypes.node,
  hideClose: PropTypes.bool,
  onToastClose: PropTypes.func,
  showToast: PropTypes.func,
  autoClose: PropTypes.number,
  toastPosition: PropTypes.string,
};

export default ToastContainer;
