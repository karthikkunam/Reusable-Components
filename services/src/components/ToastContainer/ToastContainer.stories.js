import React from 'react';
import ToastContainer from '.';

export default {
  title: 'ToastContainer',
  component: ToastContainer,
};

const toastHeader = (
  <div>
    <span className="icon icon-info"></span>
    Toast Header
  </div>
);

export const TopLeftToast = () => (
  <ToastContainer toastHeader={toastHeader} toastPosition="toast-top-left">
    Some text inside the toast body
  </ToastContainer>
);
export const TopRightToast = () => (
  <ToastContainer toastHeader={toastHeader} toastPosition="toast-top-right">
    Some text inside the toast body
  </ToastContainer>
);
export const BottomLeftToast = () => (
  <ToastContainer toastHeader={toastHeader} toastPosition="toast-bottom-left">
    Some text inside the toast body
  </ToastContainer>
);
export const BottomRightToast = () => (
  <ToastContainer toastHeader={toastHeader} toastPosition="toast-bottom-right">
    Some text inside the toast body
  </ToastContainer>
);
export const TopCenterToast = () => (
  <ToastContainer toastHeader={toastHeader} toastPosition="toast-top-center">
    Some text inside the toast body
  </ToastContainer>
);
export const BottomCenterToast = () => (
  <ToastContainer toastHeader={toastHeader} toastPosition="toast-bottom-center">
    Some text inside the toast body
  </ToastContainer>
);
export const ToastWithAutoClose = () => (
  <ToastContainer toastHeader={toastHeader} autoClose={8000} toastPosition="toast-bottom-center">
    It will autoClose in 8000 ms
  </ToastContainer>
);
