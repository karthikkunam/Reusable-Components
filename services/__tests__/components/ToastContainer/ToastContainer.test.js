import React from 'react';
import { shallow } from 'enzyme';
import ToastContainer from '../../../src/components/ToastContainer';

const props = {
  toastHeader: '',
  hideClose: false,
  autoClose: 5000,
  showToast: true,
  onToastClose: jest.fn(),
};
describe('ToastContainer Component', () => {
  it('renders correctly', () => {
    const component = shallow(<ToastContainer {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('mock setTimeout test', done => {
    const component = shallow(<ToastContainer {...props} />);
    setTimeout(() => {
      console.log('TIME IS UP');
      done();
    }, 5000);
    expect(component).toMatchSnapshot();
  });

  it('renders onClose properly', () => {
    const component = shallow(<ToastContainer {...props} />);
    component
      .find('div')
      .at(1)
      .find('button')
      .simulate('click', {
        onClose: () => {},
      });
    expect(component).toMatchSnapshot();
  });
});
