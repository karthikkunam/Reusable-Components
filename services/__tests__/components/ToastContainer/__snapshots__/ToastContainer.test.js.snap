// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastContainer Component mock setTimeout test 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  className="toast undefined show"
  data-autohide="true"
  data-delay="3000"
  role="alert"
>
  <div
    className="toast-header"
  >
    <strong
      className="mr-auto"
    />
    <button
      className="close"
      data-dismiss="toast"
      onClick={[Function]}
      type="button"
    >
      <span
        className="icon icon-close"
      />
    </button>
  </div>
  <div
    className="toast-body"
  />
</div>
`;

exports[`ToastContainer Component renders correctly 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  className="toast undefined show"
  data-autohide="true"
  data-delay="3000"
  role="alert"
>
  <div
    className="toast-header"
  >
    <strong
      className="mr-auto"
    />
    <button
      className="close"
      data-dismiss="toast"
      onClick={[Function]}
      type="button"
    >
      <span
        className="icon icon-close"
      />
    </button>
  </div>
  <div
    className="toast-body"
  />
</div>
`;

exports[`ToastContainer Component renders onClose properly 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  className="toast undefined hide"
  data-autohide="true"
  data-delay="3000"
  role="alert"
>
  <div
    className="toast-header"
  >
    <strong
      className="mr-auto"
    />
    <button
      className="close"
      data-dismiss="toast"
      onClick={[Function]}
      type="button"
    >
      <span
        className="icon icon-close"
      />
    </button>
  </div>
  <div
    className="toast-body"
  />
</div>
`;
